File: Python/7py 1/extra_perfect.py
"""
|--------------------------------------------------------------------------
| Extra Perfect Numbers (Special Numbers Series #7)
|--------------------------------------------------------------------------
|
"""

def extra_perfect(n):
   return [i for i in range(1,n+1,2)]


def extra_perfect(n):
   return range(1, n + 1, 2)


def extra_perfect(n):
   return [i for i in range(n+1) if i % 2]


print(extra_perfect(3), [1,3])
print(extra_perfect(5), [1,3,5])
print(extra_perfect(7), [1,3,5,7])
print(extra_perfect(28), [1,3,5,7,9,11,13,15,17,19,21,23,25,27])
print(extra_perfect(39), [1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39])

File: PHP/7php 1/extraPerfect.php
<?php
/*
|--------------------------------------------------------------------------
| Extra Perfect Numbers (Special Numbers Series #7)
|--------------------------------------------------------------------------
|
*/

function extraPerfect($n) {
   $res = [];
   for ($i=1; $i<=$n; $i+=2) array_push($res, $i);
   return $res;
}


function extraPerfect($n) {
   return range(1, $n, 2);
}


function extraPerfect($n) {
   $result = [];
   $num = 1;
   while($num <= $n) {
      array_push($result, $num);
      $num += 2;
   }
   return $result;
}


print_r(extraPerfect(3));
print_r(extraPerfect(5));
print_r(extraPerfect(7));